presto-main/src/test/java/com/facebook/presto/operator/scalar/AbstractTestFunctions.java

```diff
@@ -41,27 +43,50 @@
 import static com.facebook.presto.spi.StandardErrorCode.NUMERIC_VALUE_OUT_OF_RANGE;
 import static com.facebook.presto.spi.type.DecimalType.createDecimalType;
 import static com.facebook.presto.type.UnknownType.UNKNOWN;
+import static io.airlift.testing.Closeables.closeAllRuntimeException;
 import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
 import static org.testng.Assert.assertEquals;
 import static org.testng.Assert.fail;
 
 public abstract class AbstractTestFunctions
 {
-    protected final FunctionAssertions functionAssertions;
+    private final Session session;
+    private final FeaturesConfig config;
+    protected FunctionAssertions functionAssertions;
 
     protected AbstractTestFunctions()
     {
-        functionAssertions = new FunctionAssertions();
+        this(TEST_SESSION);
     }
 
     protected AbstractTestFunctions(Session session)
     {
-        functionAssertions = new FunctionAssertions(session);
+        this(session, new FeaturesConfig());
     }
 
-    protected AbstractTestFunctions(FeaturesConfig featuresConfig)
+    protected AbstractTestFunctions(FeaturesConfig config)
     {
-        functionAssertions = new FunctionAssertions(TEST_SESSION, featuresConfig);
+        this(TEST_SESSION, config);
+    }
+
+    protected AbstractTestFunctions(Session session, FeaturesConfig config)
+    {
+        this.session = requireNonNull(session, "session is null");
+        this.config = requireNonNull(config, "config is null");
+    }
+
+    @BeforeClass
+    public void setUp()
```

Comment:
:+1:  for `final`. If the child class has `@BeforeClass void moreSetup()`, it will be called after this method. So, is the `setupChild` explicit hook needed?